buffer/src/main/java/io/netty5/buffer/internal/CleanerPool.java

```diff
@@ -0,0 +1,200 @@
+
+    /**
+     * Static initializer used to setup Method Handles that are used to obtain loom virtual Threadfactory
+     * (if the platform supports it).
+     */
+    static {
+        MethodHandle mhOfVirtualTmp = null;
+        MethodHandle mhFactoryTmp = null;
+
+        if (USE_VTHREAD) {
+            try {
+                Class<?> clzOfVirtual = Class.forName("java.lang.Thread$Builder$OfVirtual");
+                mhOfVirtualTmp = publicLookup.findStatic(Thread.class, "ofVirtual",
+                        MethodType.methodType(clzOfVirtual));
+                mhFactoryTmp = publicLookup.findVirtual(clzOfVirtual, "factory",
+                        MethodType.methodType(ThreadFactory.class));
+            } catch (ClassNotFoundException e) {
+                logger.debug("Loom not supported, Cleaner will use default cleaner daemon threads.");
+            } catch (Throwable t) {
+                logger.warn("Could not create virtual thread factory, will use default cleaner daemon threads.", t);
+            }
+        }
+
+        mhOfVirtual = mhOfVirtualTmp;
+        mhFactory = mhFactoryTmp;
+    }
+
+    /**
+     * The singleton for the CleanerPool.
+     */
+    public static final CleanerPool INSTANCE = new CleanerPool();
+
+    /**
+     * The FastThreadLocal used to map a Cleaner instance to Event Loop threads.
+     */
+    private static class CleanerThreadLocal extends FastThreadLocal<Cleaner> {
+        private final Cleaner[] cleaners;
+        private static final AtomicInteger counter = new AtomicInteger();
+
+        CleanerThreadLocal(Cleaner[] cleaners) {
+            this.cleaners = cleaners;
+        }
+
+        @Override
+        protected Cleaner initialValue() {
+            return cleaners[counter.getAndIncrement() % POOL_SIZE];
+        }
+    }
+
+    private CleanerPool() {
+        logger.info("Instantiating CleanerPool: {}={}, {}={}, using vthreads={}",
+                        CNF_POOL_SIZE, POOL_SIZE,
+                        CNF_USE_VTHREAD, USE_VTHREAD,
+                        mhFactory != null);
+    }
+
+    /**
+     * Returns the next available cleaner to the calling thread.
+     * The same thread will get the same cleaner for every nextCleaner method calls.
+     */
+    Cleaner nextCleaner() {
+        return threadLocalCleaner.get();
+    }
+
+    private static int getPoolSize() {
+        int poolSize = SystemPropertyUtil.getInt(CNF_POOL_SIZE, 0);
+        if (poolSize < 0) {
+            throw new IllegalArgumentException(CNF_POOL_SIZE + " is negative: " + poolSize);
+        }
+        return poolSize == 0 ? Runtime.getRuntime().availableProcessors() : poolSize;
+    }
+
+    /**
+     * Creates a cleaner initialized with a virtual ThreadFactory if the platform supports loom,
+     * else creates an usual Cleaner with default cleaner daemon thread.
+     */
+    private Cleaner createCleaner() {
+        ThreadFactory virtualThreadFactory = getVirtualThreadFactory();
```

Comment:
Nit: Rather than using `null` as indicator could use a more descriptive boolean constant.

Reply:
I have used _Optional_ instead of the _null_ indicator. _Optional_ seems fine in this case, please have a look ?